modules/account-lib/src/coin/stx/contractBuilder.ts

```diff
@@ -149,25 +153,37 @@ export class ContractBuilder extends TransactionBuilder {
     return this;
   }
 
-  private parseCv(arg: ClarityValueJson): ClarityValue | undefined {
-    if (arg.val === 'none') {
-      return noneCV();
-    } else if (arg.type === 'buffer') {
-      return bufferCV(arg.val);
-    } else if (arg.type === 'tuple') {
-      if (arg.val instanceof Array) {
-        const data = {};
-        arg.val.forEach((a) => {
-          data[a.key] = this.parseCv({ type: a.type, val: a.val });
-        });
-        return tupleCV(data);
-      } else {
-        return noneCV();
-      }
-    } else if (typeof arg.val === 'string') {
```

Comment:
@wesgraham I think that if val was number, it didn't encoded and it was also working.